app/src/main/java/com/expidev/gcmapp/MainActivity.java

```diff
@@ -4,18 +4,87 @@
 import android.content.DialogInterface;
 import android.content.Intent;
 import android.os.Bundle;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.content.LocalBroadcastManager;
 import android.support.v7.app.ActionBarActivity;
+import android.util.Log;
 import android.view.Menu;
 import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.expidev.gcmapp.GcmTheKey.GcmBroadcastReceiver;
+import com.expidev.gcmapp.GcmTheKey.GcmTheKeyHelper;
+import com.expidev.gcmapp.http.GcmApiClient;
+import com.expidev.gcmapp.http.TokenTask;
+import com.expidev.gcmapp.model.User;
+import com.expidev.gcmapp.utils.Device;
+import com.expidev.gcmapp.utils.GcmProperties;
+
+import org.json.JSONObject;
+
+import java.util.Properties;
+
+import me.thekey.android.TheKey;
+import me.thekey.android.lib.TheKeyImpl;
+import me.thekey.android.lib.support.v4.dialog.LoginDialogFragment;
 
 
 public class MainActivity extends ActionBarActivity
 {
+    private final String TAG = this.getClass().getSimpleName();
+    private Properties properties;
+    private TheKey theKey;
+    private long keyClientId;
+
     @Override
     protected void onCreate(Bundle savedInstanceState)
     {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        getProperties();
+
+        keyClientId = Long.parseLong(properties.getProperty("TheKeyClientId", ""));
+
+        theKey = TheKeyImpl.getInstance(this, keyClientId);
+
+        LocalBroadcastManager manager = LocalBroadcastManager.getInstance(getApplicationContext());
+        GcmBroadcastReceiver gcmBroadcastReceiver = new GcmBroadcastReceiver();
+        gcmBroadcastReceiver.registerReceiver(manager);
```

Comment:
You are technically leaking the GcmBroadcastReceiver since you aren't unregistering it when the activity is destroyed. 
Memory-wise this isn't a bad leak since it doesn't hold any context references, but if the activity is recreated for various reasons you will end up with multiple GcmBroadcastReceivers registered and start seeing multiple callbacks on login/logout.